fairmq/tools/CppSTL.h:27:31: error: no matching function for call ...

https://cdash.gsi.de/testDetails.php?test=7150233&build=246481

```
     183    In file included from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/plugins/PMIx/PMIxCommands.h:16,
     184                     from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/plugins/PMIx/PMIxPlugin.h:13,
     185                     from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/plugins/PMIx/PMIxPlugin.cxx:9:
     186    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/tools/CppSTL.h: In instantiation of 'std::unique_ptr<_Tp> fair::mq::tools::make_unique(Args&& ...) [with T = fair::mq::sdk::cmd::TransitionStatus; Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, int, fair::mq::sdk::cmd::Result, fair::mq::Transition&}]':
     187    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/sdk/commands/Commands.h:357:43:   required from 'std::unique_ptr<fair::mq::sdk::cmd::Cmd> fair::mq::sdk::cmd::make(Args&& ...) [with C = fair::mq::sdk::cmd::TransitionStatus; Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, int, fair::mq::sdk::cmd::Result, fair::mq::Transition&}]'
     188    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/plugins/PMIx/PMIxPlugin.cxx:151:93:   required from here
  >> 189    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/tools/CppSTL.h:27:31: error: no matching function for call to 'fair::mq::sdk::cmd::TransitionStatus::TransitionStatus(std::__cxx11::basic_string<char>&, int, fair::mq::sdk::cmd::Result, fair::mq::Transition&)'
     190       27 |     return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
     191          |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     192    In file included from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/plugins/PMIx/PMIxPlugin.cxx:11:
     193    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.s1s/stage/alfaci/spack-stage-fairmq-develop-z2uz6uxptrf2hdxvk26y4jmqcxych5b6/spack-src/fairmq/sdk/commands/Commands.h:191:14: note: candidate: 'fair::mq::sdk::cmd::TransitionStatus::TransitionStatus(const string&, uint64_t,fair::mq::sdk::cmd::Result, fair::mq::Transition, fair::mq::State)'
     194      191 |     explicit TransitionStatus(const std::string& deviceId, const uint64_t taskId, const Result result, const Transition transition, State currentState)
     195          |              ^~~~~~~~~~~~~~~~
```

https://cdash.gsi.de/testDetails.php?test=7150214&build=246480

```
     183    In file included from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.
            lTd/stage/alfaci/spack-stage-fairmq-develop-t4664eloy5z5epzsw3fx6ki
            y2jfkyuah/spack-src/fairmq/plugins/PMIx/PMIxCommands.h:16,
     184                     from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.
            lTd/stage/alfaci/spack-stage-fairmq-develop-t4664eloy5z5epzsw3fx6ki
            y2jfkyuah/spack-src/fairmq/plugins/PMIx/PMIxPlugin.h:13,
     185                     from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.
            lTd/stage/alfaci/spack-stage-fairmq-develop-t4664eloy5z5epzsw3fx6ki
            y2jfkyuah/spack-src/fairmq/plugins/PMIx/PMIxPlugin.cxx:9:
     186    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.lTd/stage/alfaci/spack
            -stage-fairmq-develop-t4664eloy5z5epzsw3fx6kiy2jfkyuah/spack-src/fa
            irmq/tools/CppSTL.h: In instantiation of 'std::unique_ptr<_Tp> fair
            ::mq::tools::make_unique(Args&& ...) [with T = fair::mq::sdk::cmd::
            TransitionStatus; Args = {std::__cxx11::basic_string<char, std::cha
            r_traits<char>, std::allocator<char> >&, int, fair::mq::sdk::cmd::R
            esult, fair::mq::Transition&}]':
     187    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.lTd/stage/alfaci/spack
            -stage-fairmq-develop-t4664eloy5z5epzsw3fx6kiy2jfkyuah/spack-src/fa
            irmq/sdk/commands/Commands.h:357:43:   required from 'std::unique_p
            tr<fair::mq::sdk::cmd::Cmd> fair::mq::sdk::cmd::make(Args&& ...) [w
            ith C = fair::mq::sdk::cmd::TransitionStatus; Args = {std::__cxx11:
            :basic_string<char, std::char_traits<char>, std::allocator<char> >&
            , int, fair::mq::sdk::cmd::Result, fair::mq::Transition&}]'
     188    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.lTd/stage/alfaci/spack
            -stage-fairmq-develop-t4664eloy5z5epzsw3fx6kiy2jfkyuah/spack-src/fa
            irmq/plugins/PMIx/PMIxPlugin.cxx:151:93:   required from here
  >> 189    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.lTd/stage/alfaci/spack
            -stage-fairmq-develop-t4664eloy5z5epzsw3fx6kiy2jfkyuah/spack-src/fa
            irmq/tools/CppSTL.h:27:31: error: no matching function for call to 
            'fair::mq::sdk::cmd::TransitionStatus::TransitionStatus(std::__cxx1
            1::basic_string<char>&, int, fair::mq::sdk::cmd::Result, fair::mq::
            Transition&)'
     190       27 |     return std::unique_ptr<T>(new T(std::forward<Args>(args
            )...));
     191          |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ~~~~~
     192    In file included from /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.
            lTd/stage/alfaci/spack-stage-fairmq-develop-t4664eloy5z5epzsw3fx6ki
            y2jfkyuah/spack-src/fairmq/plugins/PMIx/PMIxPlugin.cxx:11:
     193    /tmp/jenkins-FairRootGroup-FairSoft-PR-318-1.lTd/stage/alfaci/spack
            -stage-fairmq-develop-t4664eloy5z5epzsw3fx6kiy2jfkyuah/spack-src/fa
            irmq/sdk/commands/Commands.h:191:14: note: candidate: 'fair::mq::sd
            k::cmd::TransitionStatus::TransitionStatus(const string&, uint64_t,
             fair::mq::sdk::cmd::Result, fair::mq::Transition, fair::mq::State)
            '
     194      191 |     explicit TransitionStatus(const std::string& deviceId, 
            const uint64_t taskId, const Result result, const Transition transi
            tion, State currentState)
     195          |              ^~~~~~~~~~~~~~~~
```

--- a/fairmq/plugins/PMIx/PMIxPlugin.cxx
+++ b/fairmq/plugins/PMIx/PMIxPlugin.cxx
@@ -148,12 +148,12 @@ auto PMIxPlugin::SubscribeForCommands() -> void
                     Transition transition = static_cast<ChangeState&>(*cmd).GetTransition();
                     if (ChangeDeviceState(transition)) {
                         fCommands.Send(
-                            Cmds(make<TransitionStatus>(fDeviceId, 0, Result::Ok, transition))
+                            Cmds(make<TransitionStatus>(fDeviceId, 0, Result::Ok, transition, GetCurrentDeviceState()))
                                 .Serialize(Format::JSON),
                             {sender});
                     } else {
                         fCommands.Send(
-                            Cmds(make<TransitionStatus>(fDeviceId, 0, Result::Failure, transition))
+                            Cmds(make<TransitionStatus>(fDeviceId, 0, Result::Failure, transition, GetCurrentDeviceState()))
                                 .Serialize(Format::JSON),
                             {sender});
                     }

--- a/test/memory_resources/_memory_resources.cxx
+++ b/test/memory_resources/_memory_resources.cxx
@@ -6,67 +6,83 @@
  *                  copied verbatim in the file "LICENSE"                       *
  ********************************************************************************/
 
-#include <cstring>
 #include <fairmq/FairMQTransportFactory.h>
 #include <fairmq/MemoryResourceTools.h>
+#include <fairmq/ProgOptions.h>
+#include <fairmq/Tools.h>
+
 #include <boost/container/pmr/polymorphic_allocator.hpp>
+
 #include <gtest/gtest.h>
+
+#include <cstring>
 #include <vector>
 
-namespace {
+namespace
+{
 
 using namespace std;
 using namespace fair::mq;
-using factoryType = std::shared_ptr<FairMQTransportFactory>;
 
-factoryType factoryZMQ = FairMQTransportFactory::CreateTransportFactory("zeromq");
-factoryType factorySHM = FairMQTransportFactory::CreateTransportFactory("shmem");
+using FactoryType = shared_ptr<FairMQTransportFactory>;
 
-struct testData
+struct TestData
 {
     int i{1};
     static int nallocated;
     static int nallocations;
     static int ndeallocations;
-    testData()
+
+    TestData()
     {
         ++nallocated;
         ++nallocations;
     }
-    testData(const testData& in)
+
+    TestData(const TestData& in)
         : i{in.i}
     {
         ++nallocated;
         ++nallocations;
     }
-    testData(const testData&& in)
+
+    TestData(const TestData&& in)
         : i{in.i}
     {
         ++nallocated;
         ++nallocations;
     }
-    testData(int in)
+
+    TestData(int in)
         : i{in}
     {
         ++nallocated;
         ++nallocations;
     }
-    ~testData()
+
+    ~TestData()
     {
         --nallocated;
         ++ndeallocations;
     }
 };
 
-int testData::nallocated = 0;
-int testData::nallocations = 0;
-int testData::ndeallocations = 0;
+int TestData::nallocated = 0;
+int TestData::nallocations = 0;
+int TestData::ndeallocations = 0;
 
-auto allocZMQ = factoryZMQ -> GetMemoryResource();
-auto allocSHM = factorySHM -> GetMemoryResource();
-
-TEST(MemoryResources, transportallocatormap)
+TEST(MemoryResources, transportAllocatorMap)
 {
+    size_t session{tools::UuidHash()};
+    ProgOptions config;
+    config.SetProperty<string>("session", to_string(session));
+
+    FactoryType factoryZMQ = FairMQTransportFactory::CreateTransportFactory("zeromq", fair::mq::tools::Uuid(), &config);
+    FactoryType factorySHM = FairMQTransportFactory::CreateTransportFactory("shmem", fair::mq::tools::Uuid(), &config);
+
+    auto allocZMQ = factoryZMQ->GetMemoryResource();
+    auto allocSHM = factorySHM->GetMemoryResource();
+
     EXPECT_TRUE(allocZMQ != nullptr && allocSHM != allocZMQ);
     auto _tmp = factoryZMQ->GetMemoryResource();
     EXPECT_TRUE(_tmp == allocZMQ);
@@ -76,28 +92,45 @@ using namespace fair::mq::pmr;
 
 TEST(MemoryResources, allocator)
 {
-    testData::nallocations = 0;
-    testData::ndeallocations = 0;
+    TestData::nallocations = 0;
+    TestData::ndeallocations = 0;
+
+    size_t session{tools::UuidHash()};
+    ProgOptions config;
+    config.SetProperty<string>("session", to_string(session));
+
+    FactoryType factoryZMQ = FairMQTransportFactory::CreateTransportFactory("zeromq", fair::mq::tools::Uuid(), &config);
+
+    auto allocZMQ = factoryZMQ->GetMemoryResource();
 
     {
-        std::vector<testData, polymorphic_allocator<testData>> v(polymorphic_allocator<testData>{allocZMQ});
+        std::vector<TestData, polymorphic_allocator<TestData>> v(polymorphic_allocator<TestData>{allocZMQ});
         v.reserve(3);
         EXPECT_TRUE(v.capacity() == 3);
         EXPECT_TRUE(allocZMQ->getNumberOfMessages() == 1);
         v.emplace_back(1);
         v.emplace_back(2);
         v.emplace_back(3);
-        EXPECT_TRUE((fair::mq::byte*)&(*v.end()) - (fair::mq::byte*)&(*v.begin()) == 3 * sizeof(testData));
-        EXPECT_TRUE(testData::nallocated == 3);
+        EXPECT_TRUE((fair::mq::byte*)&(*v.end()) - (fair::mq::byte*)&(*v.begin()) == 3 * sizeof(TestData));
+        EXPECT_TRUE(TestData::nallocated == 3);
     }
-    EXPECT_TRUE(testData::nallocated == 0);
-    EXPECT_TRUE(testData::nallocations == testData::ndeallocations);
+    EXPECT_TRUE(TestData::nallocated == 0);
+    EXPECT_TRUE(TestData::nallocations == TestData::ndeallocations);
 }
 
 TEST(MemoryResources, getMessage)
 {
-    testData::nallocations = 0;
-    testData::ndeallocations = 0;
+    TestData::nallocations = 0;
+    TestData::ndeallocations = 0;
+
+    size_t session{tools::UuidHash()};
+    ProgOptions config;
+    config.SetProperty<string>("session", to_string(session));
+
+    FactoryType factoryZMQ = FairMQTransportFactory::CreateTransportFactory("zeromq", fair::mq::tools::Uuid(), &config);
+    FactoryType factorySHM = FairMQTransportFactory::CreateTransportFactory("shmem", fair::mq::tools::Uuid(), &config);
+
+    auto allocZMQ = factoryZMQ->GetMemoryResource();
 
     FairMQMessagePtr message{nullptr};
 
@@ -105,7 +138,7 @@ TEST(MemoryResources, getMessage)
 
     // test message creation on the same channel it was allocated with
     {
-        std::vector<testData, polymorphic_allocator<testData>> v(polymorphic_allocator<testData>{allocZMQ});
+        std::vector<TestData, polymorphic_allocator<TestData>> v(polymorphic_allocator<TestData>{allocZMQ});
         v.emplace_back(1);
         v.emplace_back(2);
         v.emplace_back(3);
@@ -114,13 +147,13 @@ TEST(MemoryResources, getMessage)
         EXPECT_TRUE(message != nullptr);
         EXPECT_TRUE(message->GetData() == vectorBeginPtr);
     }
-    EXPECT_TRUE(message->GetSize() == 3 * sizeof(testData));
+    EXPECT_TRUE(message->GetSize() == 3 * sizeof(TestData));
     messageArray = static_cast<int*>(message->GetData());
     EXPECT_TRUE(messageArray[0] == 1 && messageArray[1] == 2 && messageArray[2] == 3);
 
     // test message creation on a different channel than it was allocated with
     {
-        std::vector<testData, polymorphic_allocator<testData>> v(polymorphic_allocator<testData>{allocZMQ});
+        std::vector<TestData, polymorphic_allocator<TestData>> v(polymorphic_allocator<TestData>{allocZMQ});
         v.emplace_back(4);
         v.emplace_back(5);
         v.emplace_back(6);
@@ -130,7 +163,7 @@ TEST(MemoryResources, getMessage)
         EXPECT_TRUE(message->GetData() != vectorBeginPtr);
     }
 
-    EXPECT_TRUE(message->GetSize() == 3 * sizeof(testData));
+    EXPECT_TRUE(message->GetSize() == 3 * sizeof(TestData));
     messageArray = static_cast<int*>(message->GetData());
     EXPECT_TRUE(messageArray[0] == 4 && messageArray[1] == 5 && messageArray[2] == 6);
 }

